src/pykeen/nn/pyg.py

```diff
@@ -0,0 +1,391 @@
+            message passing layer.
+
+        :param activation:
+            the activation(s), or hints thereof
+        :param activation_kwargs:
+            additional keyword-based parameters passed to the activations upon instantiation
+
+        :param kwargs:
+            additional keyword-based parameters passed to :meth:`Representation.__init__`
+
+        :raises ImportError:
+            if PyTorch Geometric is not installed
+        :raises ValueError:
+            if the number of activations and message passing layers do not match (after input normalization)
+        """
+        # fail if dependencies are missing
+        if MessagePassing is None or layer_resolver is None:
+            raise ImportError(_PYG_INSTALLATION_TEXT)
+
+        # avoid cyclic import
+        from . import representation_resolver
+
+        # the base representations, e.g., entity embeddings or features
+        base = representation_resolver.make(base, pos_kwargs=base_kwargs, max_id=triples_factory.num_entities)
+
+        super().__init__(max_id=kwargs.pop("max_id", base.max_id), shape=output_shape or base.shape, **kwargs)
+
+        # assign sub-module *after* super call
+        self.base = base
+
+        # initialize layers
+        self.layers = nn.ModuleList(layer_resolver.make_many(layers, layers_kwargs))
+
+        # normalize activation
+        activation = list(upgrade_to_sequence(activation))
+        if len(activation) == 1:
+            activation = activation * len(self.layers)
+        self.activations = nn.ModuleList(activation_resolver.make_many(activation, activation_kwargs))
+        if len(self.layers) != len(self.activations):
+            raise ValueError(
+                f"The lengths of the list of message passing layers ({len(self.layers)}) "
+                f"and activation layers ({len(self.activation)}) differs! To disable activations "
+                f"on certain layers, e.g., the last, use torch.nn.Identity."
+            )
+
+        # prepare buffer
+        # TODO: inductive?
+        self.register_buffer(name="edge_index", tensor=triples_factory.mapped_triples[:, [0, 2]].t())
+
+    def _plain_forward(self, indices: Optional[torch.LongTensor] = None) -> torch.FloatTensor:  # noqa: D102
+        # TODO: we could reduce the memory footprint and maybe also computation time
+        #       by considering only the k-hop neighborhood of the requested indices
+        #       computing this may be computationally expensive, too
+        # get *all* base representations
+        x = self.base(indices=None)
+        # perform message passing on *all* base representations & edges
+        x = self._message_passing(x=x)
+        # select desired indices
+        if indices is not None:
+            x = x[indices]
+        return x
+
+    @abstractmethod
+    def _message_passing(self, x: torch.FloatTensor) -> torch.FloatTensor:
```

Comment:
wouldn't `pass_message` be more descriptive?

Reply:
This function implements essentially
https://github.com/pykeen/pykeen/blob/ba3c72adc924c34d4ee3cb4609f2519009cf6f20/src/pykeen/nn/pyg.py#L249-L252
with some modifications to the parameters the individual message passing layers receives.

So it applies a sequence of message passing layers with activation functions in between to the tensor of base representations `x`.

So maybe `pass_messages`, or `apply_message_passing`?

Reply:
`pass_messages`, I typod earlier :)

Reply:
Done in ba3c72ad

Reply:
Message passing is a very common term in the GNN literature, it's standard in PyG and DGL, so from that perspective keeping `_message_passing` would be more readable and click faster for the GNN folks :)